sources/advertisements/VASTParser.swift

```diff
@@ -491,6 +506,27 @@ enum VASTParser {
         
         return delegateStack
     }
+    
+    static func parseTime(from string: String) -> Double? {
+        let components = string.components(separatedBy: ":")
+        guard components.count == 3 else { return nil }
+        let secondsArray: [Double] = components
+            .compactMap(Double.init)
+            .enumerated()
+            .compactMap {
+                let multipliers: [Double] = [3600, 60, 1]
+                let maxValues: [Double] = [24, 60, 60]
+                let floatingValues = [false, false, true]
```

Comment:
Looks like this is new data struct. You can create `strcut` with all this rules  for each time component and put it into the array